### DISCUSSION TITLE
Image quality metrics of multi-echo datasets -- where to go?

### DISCUSSION THREAD

**Discussion by oesteban** [nipreps/mriqc#1335] [Ideas]
MRIQC had long marginalized multi-echo datasets. Originally, it would treat them as different BOLD runs, and that's it.

This presented us with challenges and inconsistencies. One very obvious is that between-volume head motion should be *the same* through every echo at each timestep. However, our original approach would calculate head motion per echo and it obviously would not match across echos. More worryingly, performing image registration with long echos with low SNR and great dropout from distortions is very unreliable.

That made us change the view and now MRIQC treats all echos in a multiecho dataset as part of a single run. There are many wrinkles to iron out here and there (e.g., #1334), but the most challenging issue is to generate a single set of IQMs that make the most sense for ME.

@garciaml participated in the NiPreps Hack and worked on this topic. She has developed a notebook (https://github.com/garciaml/mriqc_fMRI_multi-echos_iqms) showing what metrics can be more predictive of specific artifacts and how they relate.

With her hands-on experience and the me-experts (looking at @emdupre, @eilidhmacnicol, @cesarcaballerogaudes, and laterally, @tsalo), I'd like to address the question: **what quality metrics should we define within MRIQC**?

So far, we have reached the conclusion that FD (framewise displacement) should be a single value per timepoint across echos, and we are running image registration on the shortest echo (which has the largest SNR and t1 contrast), on top of which the FD is calculated.

An appealing solution could be to calculate the combined echoes and then run standard se-bold metrics. However, I'm weary that this is *too much processing* for MRIQC, which should be as attached as possible to the original unprocessed data.

We could start moving by averaging across echos all the different metrics (except FD), but I'm not sure this is going to be consistent across me options (e.g., 3 echos vs. 6 echos).

Thoughts?

/cc @celprov

**Comment by oesteban** [comment]
BTW, there's an issue open for this  #1184, generalizing from #1183.

**Comment by BastienCagna** [comment]
Hi @oesteban,
Is there any update on this? 
Is taking only one echo (the nearest to usual single echo time) possible ?
The link to @garciadias is dead.

**Reply by garciaml** [reply]
Hi,

Apologies I wasn't super available for a couple of months.
Thank you for the reminder!

Short answer: No. (from my little experience on this)

Certain artifacts are more visible on short echoes and others on longer echoes due to physical properties.

I studied various simple strategies for combining IQM values across echoes, and got artifact-specific (and phase-encoding direction - specific) sets of associated features and machine learning models. 

Notably, on the dataset I used, several IQMs appeared completely independent of echo time: FD mean, FD num, FD perc, summary BG n, summary FG n. So you may use these on one echo of your choice (FD to assess head motion for instance). 

I am currently finishing the manuscript on this study (mainly discussion part), but I would be very open to meet and share my predictive models with you if you want to test those on your dataset. Please let me know if you would like to meet and I'll share my contact info privately.

Thank you for your patience, wish you a great day!

Mélanie

**Reply by BastienCagna** [reply]
Ok, Thanks Mélanie for your responsivness and your insights!
We are starting to works on mte data and was wondering on how to interperet some IQMs (like tSNR). We are thinking to rerun MRIQC on preprocessed data (which include combine without denoising). We have also a lot to do on our side and may not be available to spend a lot of time on this right now but keeping you in touch if it does. Thanks!
Bastien

**Reply by garciaml** [reply]
In my results, I can see that features derived from tSNR showed distribution shift on the presence of head motion artifact, and were important features to predict head motion in the ML models too. It included the max value (echo 1), median, range, and the echo 3 value (TE=30ms / to be confirmed by @celprov or oe). Distribution shift was also observed for intensity non-uniformity artifact for several features derived from tSNR, but not found as important predictors in the ML analysis. No distribution shift was observed on global noise, wraparound and ghosting artifacts. 
Again, this is relative to the dataset I used, and I am very curious to know if these observations replicate on new datasets.
Happy to keep in touch!

**Comment by garciadias** [comment]
Curiously enough, I have worked on this project, but I think you tagged the wrong person, @BastienCagna. I see you are already getting your response, so I will leave the conversation. 👋

**Reply by BastienCagna** [reply]
Oh yes, I wrote the message too quickly, sorry.
